codecov_cli/parsers/junit.py

```diff
@@ -0,0 +1,51 @@
+import logging
+from datetime import datetime, timedelta
+from typing import List
+
+from lxml import etree
+
+from codecov_cli.parsers.base import Parser, ParsingError, Test, Testrun
+
+logger = logging.getLogger("codecovcli")
+
+
+class JUnitXMLParser(Parser):
+    def __init__(self):
+        self._parser = etree.XMLParser(recover=True, resolve_entities=False)
+
+    def parse(self, file_content) -> List[Testrun]:
+        processed = self._parse_xml(file_content)
+        if processed is None or len(processed) == 0:
+            raise ParsingError("Error parsing XML file")
+
+        testsuites = [
+            self._create_testsuite(testsuite_xml)
+            for testsuite_xml in processed.iter("testsuite")
+        ]
+
+        return testsuites
+
+    def _create_testcase(self, testcase_xml: etree.Element):
+        return Test(
+            f"{testcase_xml.get('classname')}.{testcase_xml.get('name')}",
```

Comment:
Can this be transformed in the format that we use for labels?
That is `path/to/test.py::TestClassIfAny::test_function_name`.

It would be helpful to use this in an ATS setting to have the test names follow this format.